Upgrade to Heroku-22

This is currently running on the Heroku-18 stack, which is deprecated and will start failing to build in February, 2023. 

Instructions for upgrading are at https://devcenter.heroku.com/articles/upgrading-to-the-latest-stack#general-upgrading-strategy

We should test any upgrade on RC before upgrading production.

Upgraded python to 3.9, but heroku-22 now fails with the following:

```
         Building wheel for xmlsec (setup.py): started
         Building wheel for xmlsec (setup.py): finished with status 'error'
         error: subprocess-exited-with-error
         
         × python setup.py bdist_wheel did not run successfully.
         │ exit code: 1
         ╰─> [52 lines of output]
             /app/.heroku/python/lib/python3.9/site-packages/setuptools/dist.py:739: UserWarning: Usage of dash-separated 'description-file' will not be supported in future versions. Please use the underscore name 'description_file' instead
               warnings.warn(
             /app/.heroku/python/lib/python3.9/site-packages/setuptools/dist.py:739: UserWarning: Usage of dash-separated 'build-requires' will not be supported in future versions. Please use the underscore name 'build_requires' instead
               warnings.warn(
             /app/.heroku/python/lib/python3.9/site-packages/setuptools/dist.py:739: UserWarning: Usage of dash-separated 'upload-dir' will not be supported in future versions. Please use the underscore name 'upload_dir' instead
               warnings.warn(
             /app/.heroku/python/lib/python3.9/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
               warnings.warn(
             running bdist_wheel
             running build
             running build_ext
             building 'xmlsec' extension
             creating build
             creating build/temp.linux-x86_64-3.9
             creating build/temp.linux-x86_64-3.9/tmp
             creating build/temp.linux-x86_64-3.9/tmp/pip-install-hrjyxzlz
             creating build/temp.linux-x86_64-3.9/tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f
             creating build/temp.linux-x86_64-3.9/tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src
             gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DMODULE_NAME=xmlsec -DMODULE_VERSION=1.3.3 -D__XMLSEC_FUNCTION__=__func__ -DXMLSEC_NO_SIZE_T -DXMLSEC_NO_GOST=1 -DXMLSEC_NO_GOST2012=1 -DXMLSEC_NO_CRYPTO_DYNAMIC_LOADING=1 -DXMLSEC_CRYPTO_OPENSSL=1 -I/tmp/build_bca3d7f1/.apt/usr/include/xmlsec1 -I/usr/include/libxml2 -I/tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/.eggs/lxml-4.9.1-py3.9-linux-x86_64.egg/lxml/includes -I/tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/.eggs/lxml-4.9.1-py3.9-linux-x86_64.egg/lxml -I/tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/.eggs/lxml-4.9.1-py3.9-linux-x86_64.egg/lxml/includes/extlibs -I/tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/.eggs/lxml-4.9.1-py3.9-linux-x86_64.egg/lxml/includes/libxslt -I/tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/.eggs/lxml-4.9.1-py3.9-linux-x86_64.egg/lxml/includes/libxml -I/tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/.eggs/lxml-4.9.1-py3.9-linux-x86_64.egg/lxml/includes/libexslt -I/app/.heroku/python/include/python3.9 -c /tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src/constants.c -o build/temp.linux-x86_64-3.9/tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src/constants.o -g -std=c99 -fno-strict-aliasing -Wno-error=declaration-after-statement -Werror=implicit-function-declaration -Os
             In file included from /tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src/constants.c:10:
             /tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src/constants.c: In function ‘PyXmlSec_ConstantsModule_Init’:
             /tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src/constants.c:278:50: error: ‘xmlSecXkmsNs’ undeclared (first use in this function); did you mean ‘xmlSecXPathNs’?
               278 |     tmp = PyString_FromString((const char*)(JOIN(xmlSec, name))); \
                   |                                                  ^~~~~~
             /tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src/common.h:21:23: note: in definition of macro ‘DO_JOIN2’
                21 | #define DO_JOIN2(X,Y) X##Y
                   |                       ^
             /tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src/common.h:19:19: note: in expansion of macro ‘DO_JOIN1’
                19 | #define JOIN(X,Y) DO_JOIN1(X,Y)
                   |                   ^~~~~~~~
             /tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src/constants.c:278:45: note: in expansion of macro ‘JOIN’
               278 |     tmp = PyString_FromString((const char*)(JOIN(xmlSec, name))); \
                   |                                             ^~~~
             /tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src/constants.c:288:5: note: in expansion of macro ‘PYXMLSEC_ADD_NS_CONSTANT’
               288 |     PYXMLSEC_ADD_NS_CONSTANT(XkmsNs, "XKMS");
                   |     ^~~~~~~~~~~~~~~~~~~~~~~~
             /tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src/constants.c:278:50: note: each undeclared identifier is reported only once for each function it appears in
               278 |     tmp = PyString_FromString((const char*)(JOIN(xmlSec, name))); \
                   |                                                  ^~~~~~
             /tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src/common.h:21:23: note: in definition of macro ‘DO_JOIN2’
                21 | #define DO_JOIN2(X,Y) X##Y
                   |                       ^
             /tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src/common.h:19:19: note: in expansion of macro ‘DO_JOIN1’
                19 | #define JOIN(X,Y) DO_JOIN1(X,Y)
                   |                   ^~~~~~~~
             /tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src/constants.c:278:45: note: in expansion of macro ‘JOIN’
               278 |     tmp = PyString_FromString((const char*)(JOIN(xmlSec, name))); \
                   |                                             ^~~~
             /tmp/pip-install-hrjyxzlz/xmlsec_f949f1134b4f4c5088e656f6c291ea0f/src/constants.c:288:5: note: in expansion of macro ‘PYXMLSEC_ADD_NS_CONSTANT’
               288 |     PYXMLSEC_ADD_NS_CONSTANT(XkmsNs, "XKMS");
                   |     ^~~~~~~~~~~~~~~~~~~~~~~~
             error: command '/usr/bin/gcc' failed with exit code 1
             [end of output]
         
         note: This error originates from a subprocess, and is likely not a problem with pip.
         ERROR: Failed building wheel for xmlsec
         Running setup.py clean for xmlsec
```

--- a/Aptfile
+++ b/Aptfile
@@ -1,1 +1,3 @@
+pkg-config
+libffi-dev
 libxmlsec1-dev

--- a/apt.txt
+++ b/apt.txt
@@ -4,4 +4,6 @@ curl
 libjpeg-dev
 zlib1g-dev
 net-tools
+pkg-config
+libffi-dev
 libxmlsec1-dev

--- a/requirements.in
+++ b/requirements.in
@@ -47,7 +47,7 @@ praw==4.5.1
 pyyaml==5.4
 python-dateutil
 python-rapidjson==1.8
-python3-saml==1.9.0
+python3-saml==1.14.0
 pygithub==1.44.1
 pyopenssl==17.5.0
 pytube==9.5.2

--- a/requirements.txt
+++ b/requirements.txt
@@ -90,7 +90,6 @@ defusedxml==0.6.0
     # via
     #   cairosvg
     #   python3-openid
-    #   python3-saml
     #   social-auth-core
     #   zeep
 deprecated==1.2.7
@@ -209,7 +208,7 @@ ipython==7.16.3
     # via -r requirements.in
 ipython-genutils==0.2.0
     # via traitlets
-isodate==0.6.0
+isodate==0.6.1
     # via
     #   python3-saml
     #   zeep
@@ -226,6 +225,7 @@ kombu==5.2.4
 lxml==4.6.5
     # via
     #   -r requirements.in
+    #   python3-saml
     #   xbundle
     #   xmlsec
     #   zeep
@@ -257,8 +257,6 @@ pillow==9.0.0
     # via
     #   -r requirements.in
     #   cairosvg
-pkgconfig==1.3.1
-    # via xmlsec
 praw==4.5.1
     # via -r requirements.in
 prawcore==0.10.1
@@ -308,7 +306,7 @@ python-rapidjson==1.8
     # via -r requirements.in
 python3-openid==3.1.0
     # via social-auth-core
-python3-saml==1.9.0
+python3-saml==1.14.0
     # via -r requirements.in
 pytube==9.5.2
     # via -r requirements.in
@@ -447,7 +445,7 @@ wrapt==1.14.1
     #   deprecated
 xbundle==0.3.1
     # via -r requirements.in
-xmlsec==1.3.3
+xmlsec==1.3.13
     # via python3-saml
 zeep==3.4.0
     # via mit-moira

